apis/core/v1beta1/openstackcontrolplane_types.go

```diff
@@ -34,6 +34,10 @@ type OpenStackControlPlaneSpec struct {
 	// StorageClass -
 	StorageClass string `json:"storageClass,omitempty"`
 
+	// +kubebuilder:validation:Optional
+	// NodeSelector to target subset of worker nodes running control plane services
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
Maybe we should add a temporary comment here that indicates that this only currently works for Keystone?  Otherwise I think this patch is ready

Reply:
I can do that. MariaDB doesn't support it yet. And Rabbit will work differently. Need to have discussion on that one.

Reply:
@abays comment added